data/external/scrapers/nat.py

```diff
@@ -0,0 +1,168 @@
+# This script takes care of downloading data from the school of natural sciences (oiginally the physics department)
+import json
+from multiprocessing.pool import ThreadPool
+
+import tqdm
+from external.scraping_utils import _cached_json, _download_file, _write_cache_json, CACHE_PATH
+
+NAT_API_URL = "https://api.srv.nat.tum.de/api/v1/rom"
+NAT_CACHE_DIR = CACHE_PATH / "nat"
+
+
+def scrape_buildings():
+    """
+    Retrieve the buildings as in the NAT roomfinder.
+    """
+    cache_name = "buildings_nat.json"
+
+    buildings = _cached_json(cache_name)
+    if buildings is not None:
+        return buildings
+
+    _download_file(f"{NAT_API_URL}/building", CACHE_PATH / cache_name)
+
+    return _cached_json(cache_name)
+
+
+def scrape_rooms():
+    """
+    Retrieve the rooms as in the NAT roomfinder.
+
+    :returns: A list of rooms, each room is a dict
+    """
+    cache_name = "rooms_nat.json"
+
+    rooms = _cached_json(cache_name)
+    if rooms is not None:
+        return rooms
+
+    base_info = _get_base_room_infos()
+    rooms = {}
+    with ThreadPool() as pool:
+        for room in tqdm.tqdm(
+            pool.imap_unordered(_download_and_merge_room, base_info),
+            desc="Downloaded nat rooms",
+            total=len(base_info),
+        ):
+            key = room["room_code"]  # needed, as room_code is removed in _sanitise_room
+            rooms[key] = _sanitise_room(room)
+
+    _write_cache_json(cache_name, rooms)
+    return rooms
+
+
+def _sanitise_room(room: dict):
+    """
+    Sanitise the room information
+    """
+    room.pop("room_code")
+
+    return room
```

Comment:
Jea.. originally I had more logic here (like removing the loads of null values, spaces, empty dicts/lists, ..).
Then I noticed, that the other scrapers don't do sanitization, and removed this logic.
There are a few fields, which probably could use some sanitisation (if you have a look at the scraping results, there are a few fields, which are clearly not strings but int)

That is something I wanted to discuss but forgot to highlight:
Do we want to sanitise this data here, or in the merging step?

Reply:
For me at least the approach is usually "first scrape, then process" (to avoid anything that could break the scraping, which usually takes more time. But this is not too bad here). On the other hand some basic sanitization could make sense here if it reduces the code complexity in the merging step I think.